zipkin-server/src/main/java/zipkin/server/internal/ZipkinHttpCollector.java

```diff
@@ -56,10 +61,10 @@
   private HttpHandler next;
 
   @Autowired ZipkinHttpCollector(StorageComponent storage, CollectorSampler sampler,
-    CollectorMetrics metrics) {
+    CollectorMetrics metrics, Optional<List<SpanFilter>> filters) {
```

Comment:
we don't have to inject optional right? can't we just get an empty list?

Reply:
I had this with `@Nullable` before, but I kind of like `Optional` from a readability standpoint. Happy to change it back. @rmichela, is it better to use `Optional` or `@Nullable`?

Reply:
The `Optional<T>` is required so Spring knows the constructor injected `filters` parameter is not required. Without it, the bean graph will fail to load at startup when no `SpanFilter` plugin implementations are registered with Spring.

https://stackoverflow.com/a/42135423

Reply:
The StackOverflow answer is for injecting a single bean. Spring is aware of collection types (after all, it is injecting the list of `SpanFilter` beans, not a single bean of type `List<SpanFilter>`) and will inject an empty collection if no beans of the collection type exist, as far as I can tell.